src/transformers/trainer_seq2seq.py

```diff
@@ -174,9 +174,9 @@ def prediction_step(
 
         # XXX: adapt synced_gpus for fairscale as well
         gen_kwargs = self._gen_kwargs.copy()
-        gen_kwargs["max_length"] = (
-            gen_kwargs["max_length"] if gen_kwargs.get("max_length") is not None else self.model.config.max_length
-        )
+        if gen_kwargs.get("max_length") is None and gen_kwargs.get("max_new_tokens") is None:
+            gen_kwargs["max_length"] = self.model.config.max_length
+        prompt_seq_length = 1 if self.model.config.is_encoder_decoder else 0
```

Comment:
It seems to me `Seq2SeqTrainer` will be only used for encoder-decoder models. If this is true, we shouldn't need `else 0`. 

(If a decoder-only model is possible, we will have to get the actual length of the prompt from the inputs, instead of just set it to `0`.)

Reply:
I would like to hear from @patrickvonplaten, @patil-suraj and @sgugger for this.

Reply:
+1 I think we can be confident that the Seq2Seq trainer only works for models that have `self.model.config.is_encoder_decoder = True`